src/org/opensolaris/opengrok/configuration/SuperIndexSearcher.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package org.opensolaris.opengrok.configuration;
+
+import java.util.concurrent.ExecutorService;
+import org.apache.lucene.index.IndexReader;
+import org.apache.lucene.search.IndexSearcher;
+import org.apache.lucene.search.SearcherManager;
+
+/**
+ * Wrapper class over IndexSearcher which keeps SearcherManager around so
```

Comment:
the pattern you wanted to use here is singleton, so this might be more optimized, let's leave it like this for now ...